src/python_bindings/test_bindings.py

```diff
@@ -1,14 +1,212 @@
+            {},
+            {
+                "metric": "cosine",
+                "rhs_indices": [1],
+                "parameter": 213.213111,
+                "dist_from_null_is_infinity": False,
+                "q": 123,
+                "metric_algorithm": "approx",
+                "lhs_indices": [0, 1, 2],
+            },
+        ),
+    ],
+}
+
+metric_verifier_failure = [
+    {
+        "path": "WDC_satellites.csv",
+        "options": {
+            "metric": "euclidean",
+            "rhs_indices": [1, 2],
+        },
+    },
+    {
+        "path": "TestLong.csv",
+        "options": {
+            "metric": "euclidean",
+            "q": 123,
+        },
+    },
+    {
+        "path": "TestLong.csv",
+        "options": {
+            "metric": "euclidean",
+            "metric_algorithm": "brute",
+        },
+    },
+    {
+        "path": "TestLong.csv",
+        "options": {"metric": "levenshtein", "rhs_indices": [1, 2]},
+    },
+    {"path": "WDC_satellites.csv", "options": {"metric": "levenshtein", "q": 123}},
+    {
+        "path": "WDC_satellites.csv",
+        "options": {
+            "metric": "levenshtein",
+            "metric_algorithm": "brute",
+        },
+    },
+    {"path": "TestLong.csv", "options": {"metric": "levenshtein", "rhs_indices": [1]}},
+    {"path": "TestLong.csv", "options": {"rhs_indices": [1]}},
+    {"path": "WDC_satellites.csv", "options": {"rhs_indices": [1, 2]}},
+    {"path": "WDC_satellites.csv", "options": {"q": 123}},
+    {"path": "WDC_satellites.csv", "options": {"metric_algorithm": "approx"}},
+]
+
+
+class TestPythonBindings(unittest.TestCase):
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
+
+    def _test_correct_option_setting(
+        self, algo, path, separator, has_header, options: dict
+    ):
+        testing_algo = algo()
+        testing_algo.load_data(path, separator, has_header, **options.load_options)
+        for name, value in options.execute_options.items():
+            if name == "is_null_equal_null":
+                continue
+            testing_algo.set_option(name, value)
+
+        for name, value in list(options.execute_options.items()) + list(
+            options.load_options.items()
+        ):
+            self.assertEqual(value, testing_algo.get_opts()[name])
+
+    def test_correct_load(self):
+        for algo, options in algo_correct_options.items():
+            for option in options:
+                with self.subTest(msg=f"testing {algo} correct load"):
+                    self._test_correct_option_setting(
+                        algo, option.path, ",", True, option
+                    )
+
+    def test_metric_verifier_failure_cases(self):
+        for load in metric_verifier_failure:
+            with self.subTest(msg=f"metric_verifier_load: {load}"):
+                self.assertFalse(
+                    check_metric_verifier_failure(load["path"], **load["options"])
+                )
```

Comment:
```suggestion
                with self.assertRaises(desb.ConfigurationError):
                    check_metric_verifier_failure(path, options)
```
[docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)